### PR TITLE
check demonstration version before loading

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Currently the demonstration version is always set to 1:
https://github.com/Unity-Technologies/ml-agents/blob/d4810e28d16a13efb580337eab40db7b99cd9a8c/com.unity.ml-agents/Runtime/Demonstrations/Demonstration.cs#L36

If we decide to change the version later, we can either make the loading code handle both versions, or drop support for version 1 (hopefully the former). In the meantime, we should make sure that any attempt to load a new version in old code is unsupported.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://jira.unity3d.com/browse/MLA-840

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by mmattar** [comment]
> or drop support for version 1

If we ship v1.0 of the SDK with the Demo version set to 1, doesn't this mean that we can't drop support for it? Otherwise, that would be a major breaking change.

**Comment by chriselion** [comment]
Correct. But I don't think support for it would be a major burden, we'd just need to check the version and switch which loading code we call.

File: ml-agents/mlagents/trainers/demo_loader.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
 from google.protobuf.internal.encoder import _EncodeVarint  # type: ignore
 
 
+INITIAL_POS = 33
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/demo_loader.py:25]: Does this mean we should fix INITIAL_POS for all future versions?

Comment (chriselion) [reply ml-agents/mlagents/trainers/demo_loader.py:25]: Depends - I think the current meta proto should stay as it is. But if we wanted to add more data in a v2 format, we could have a v2 metadata that's in file after the v1 metadata.

**Pull request merged**